src/model/wallet/change-credentials/complete/CompleteRequest.ts

```
@@ -0,0 +1,26 @@
+import { BuildServiceInputError } from "../../../../domain/errors/service/BuildServiceInputError";
+
+import * as validator from "../../../../utils/validator";
+
+function validate(props): boolean {
+    return (
+        validator.isBoolean(props.authByMasterKey) &&
+        validator.isEmail(props.email) &&
+        validator.isCipher(props.encryptedCredentials)
+    );
+}
+
+export class CompleteRequest {
+    public email: string;
+    public authByMasterKey: boolean;
+    public encryptedCredentials: string;
+
+    constructor(props) {
+        if (!validate(props)) {
```

Validators can raise error instead of returning Boolean. Validation code would be simpler. Just `validate(props)`.